app/commands/v2/put_link_set.rb

```diff
@@ -0,0 +1,44 @@
+module Commands
+  module V2
+    class PutLinkSet < BaseCommand
+      def call
+        validate!
+
+        content_id = link_params.fetch(:content_id)
+
+        link_set = LinkSet.find_or_initialize_by(content_id: content_id)
```

Comment:
I realise you can't use create_or_replace here because of the need to merge, but I'm still nervous about contention. At the very least we need to use `LinkSet.lock. find_or_initialize_by` and rescue from `ActiveRecord::RecordNotUnique`.